### ISSUE TITLE
Request: Facing Indicators

### ISSUE THREAD

**Issue by JLtheking2** [Varriount/fvtt-autorotate#2]
Adding facing indicators, such as that seen in the About Face mod, would be extremely helpful as well. 

As of now that’s the only reason I haven’t converted over, a couple of my PCs still don’t use top down portraits and can’t handle rotation, yet I would like to know which way they are facing.

**Comment by Varriount** [comment]
Unfortunately, this might take a while to implement. The method About Face uses to show indicators requires patching core Foundry code, which I would rather avoid if possible.

One question though - if your player's tokens don't have top-down portraits, why have them rotate? Are you using some mechanic that depends on knowing the direction a token is facing?

**Comment by JLtheking2** [comment]
Yes, I am using a homebrew rule that allows Opportunity Attacks to only trigger if the creature leaving the threat range is currently the focus of the other creature (i.e. where that creature is facing). This allows “skirmisher” type fighters like Rogues to enter into melee range, attack, then retreat without triggering Opportunity Attacks. My players love this mechanic but it relies on knowing which direction a creature is facing.

It works on physical tabletop, of course, because miniatures have a facing and we can rotate the pieces. But when we play on Foundry we have to rely on memory for this. Hence, facing indicator would be helpful.

Thanks for the suggestion. Alas, rotating character portraits also look really wonky 😅.

**Comment by Varriount** [comment]
@JLtheking2 I'm probably going to implement this as a separate module, since there are multiple uses for being able to place an underlay on a token.

(By the way, I'm curious - is the cunning action feature not allowed? That allows rogues to disengage as a bonus action)

**Comment by JLtheking2** [comment]
Thanks for the update! Looking forward to it :)

For my houserule, cunning action is definitely allowed and still very useful. But the point of that houserule is to allow characters that aren’t rogues engage in that kind of hit-and-run style of play too, which I find is fun. Taking a full action to disengage is very punishing for non-Rogue characters, and this mechanic allows pieces on the board to still move around relatively freely and dynamically to take advantage of flanking bonuses and the like. 

As we know in vanilla, once the melee PCs start engaging with the enemy melee characters, combat kinda just becomes kinda stale with no one having any reason to move. With flanking bonuses are involved, you often get into a ridiculous conga line situation with everyone forming up into a line and no one able to move for fear of triggering two opportunity attacks. My house rule helps out with that.

**Comment by Battenhatch** [comment]
This feature is very useful for GURPS gaming, especially if you are using pog-style tokens (not top-downs). I would very much like to see it implemented.

**Comment by alexremy** [comment]
Hello, 
Chipping in on the Token Facing Indicator similar to About Face, extremely useful for systems like WFRP 4e. You mentioned a separate module for this but could not find it. Can you link?
Thanks, 

a.

**Comment by Varriount** [comment]
I haven't written such a module yet - it's on my "eventually" list.

**Comment by tsansom** [comment]
Also interested in the facing direction functionality - for my homebrew flanking rules only the person behind a flanked enemy gets advantage on the attack. So being able to show which direction a character/enemy is facing on their turn would be extremely helpful for those of us that prefer framed tokens.

**Comment by alexremy** [comment]
Thanks for the answers @Varriount much appreciated!

**Comment by Orionox** [comment]
I'd be very interested in a module thatt allowed for a underlay as well.
